src/components/FlowRunTimeline.vue

```diff
@@ -236,6 +242,119 @@
     timelineGraph.value?.centerViewport()
   }
 
+  function toggleSubFlow(id: string): void {
+    const isValueVisible = expandedSubFlows.value.has(id)
+
+    if (isValueVisible) {
+      const subFlowRun = expandedSubFlows.value.get(id)
+      subFlowRun?.subscription.unsubscribe()
+      expandedSubFlows.value.delete(id)
+      return
+    }
+
+    const subFlowId = findSubFlowId(id)
+
+    if (!subFlowId) {
+      console.warn('No subflow ID found for node', id)
+      return
+    }
+
+    const subscription = useSubscription(
+      api.flowRuns.getFlowRunsTimeline,
+      [subFlowId],
+      { interval: 5000 },
+    )
+
+    const data = computed(() => {
+      return removeNullStartsAndSort(subscription.response ?? [])
+    })
+
+    expandedSubFlows.value.set(id, {
+      data,
+      subscription,
+    })
+  }
+
+  function findSubFlowId(id: string): string | null {
+    let subFlowId = graphData.value.find((node) => node.id === id)?.subFlowRunId
+
+    if (subFlowId) {
+      return subFlowId
+    }
+
+    for (const value of expandedSubFlows.value.values()) {
+      if ('value' in value.data) {
+        subFlowId = value.data.value.find((node) => node.id === id)?.subFlowRunId
+      } else {
+        subFlowId = value.data.find((node) => node.id === id)?.subFlowRunId
+      }
+
+      if (subFlowId) {
+        return subFlowId
+      }
+    }
+
+    return null
+  }
+
+  function removeNullStartsAndSort(nodes: TimelineNode[]): GraphTimelineNode[] {
+    return nodes.filter((node): node is GraphTimelineNode => isValidGraphTimelineNode(node))
```

Comment:
If `isValidGraphTimelineNode` is a typeguard already, you shouldn't need to specify `node is GraphTimelineNode`, it should be inferred.